backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -507,25 +544,77 @@ public boolean deleteCommits(
   }
 
   @Override
-  public void addDeleteDatasetVersionTags(
+  public DatasetVersion addDeleteDatasetVersionTags(
+      RepositoryDAO repositoryDAO,
+      BlobDAO blobDAO,
       MetadataDAO metadataDAO,
       boolean addTags,
-      RepositoryEntity repositoryEntity,
+      String datasetId,
       String datasetVersionId,
       List<String> tagsList,
       boolean deleteAll)
       throws ModelDBException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      session.beginTransaction();
-      CommitEntity commitEntity =
-          getCommitEntity(session, datasetVersionId, (session1 -> repositoryEntity));
+      RepositoryEntity repositoryEntity;
 
+      RepositoryIdentification.Builder repositoryIdentification =
+          RepositoryIdentification.newBuilder();
+      if (datasetId == null || datasetId.isEmpty()) {
+        CommitEntity commitEntity = session.get(CommitEntity.class, datasetVersionId);
+        if (commitEntity == null) {
+          throw new ModelDBException("DatasetVersion not found", Code.NOT_FOUND);
+        }
+
+        if (commitEntity.getRepository() != null && commitEntity.getRepository().size() > 1) {
+          throw new ModelDBException(
+              "DatasetVersion '"
+                  + commitEntity.getCommit_hash()
+                  + "' associated with multiple datasets",
+              Code.INTERNAL);
+        }
+        Long newRepoId = new ArrayList<>(commitEntity.getRepository()).get(0).getId();
+        repositoryIdentification.setRepoId(newRepoId);
+      } else {
+        repositoryIdentification.setRepoId(Long.parseLong(datasetId));
+      }
+      repositoryEntity = repositoryDAO.getRepositoryById(repositoryIdentification.build(), true);
+      addDeleteCommitTags(
```

Comment:
should this be label?

Reply:
ok verfied that the function is actually adding label , it is just named incorrectly

Reply:
Ok, Change makes sense to me. I was set `tag` because for dataset version we called it to tags.